snorkel/labeling/apply/lf_applier.py

```diff
@@ -0,0 +1,42 @@
+from itertools import chain
+from typing import Any, List, Tuple
+
+import scipy.sparse as sparse
+from tqdm import tqdm
+
+from snorkel.labeling.lf import LabelingFunction
+from snorkel.types import DataPoint, DataPoints
+
+RowData = List[Tuple[int, int, int]]
+
+
+class BaseLFApplier:
+    def __init__(self, lfs: List[LabelingFunction]) -> None:
+        self._lfs = lfs
+
+    def _matrix_from_row_data(self, labels: List[RowData]) -> sparse.csr_matrix:
+        row, col, data = zip(*chain.from_iterable(labels))
+        n, m = len(labels), len(self._lfs)
+        return sparse.csr_matrix((data, (row, col)), shape=(n, m))
+
+    def apply(self, data_points: Any, *args: Any, **kwargs: Any) -> sparse.csr_matrix:
+        raise NotImplementedError
+
+
+def apply_lfs_to_data_point(
```

Comment:
It still kills me inside that this isn't in `BaseLFApplier`....

Reply:
How about put it in the class for now, then we'll empirically test if it hurts parallelism.  *I'll* do it

Reply:
Go for it! I'm merging this in assuming you'll do that in a separate PR